auth/src/user.rs

```diff
@@ -21,16 +23,43 @@ use stripe::{
 
 #[async_trait]
 pub trait UserManagement: Send + Sync {
+    /// Create a user with the given tier
     async fn create_user(&self, name: AccountName, tier: AccountTier) -> Result<User, Error>;
+
+    /// Upgrade a user to pro using the given checkout session
     async fn upgrade_to_pro(
         &self,
         name: &AccountName,
         checkout_session_metadata: CheckoutSession,
     ) -> Result<(), Error>;
+
+    /// Change the tier for a user
     async fn update_tier(&self, name: &AccountName, tier: AccountTier) -> Result<(), Error>;
+
+    /// Get a user by their account name
     async fn get_user(&self, name: AccountName) -> Result<User, Error>;
+
+    /// Get a user by their api key
     async fn get_user_by_key(&self, key: ApiKey) -> Result<User, Error>;
+
+    /// Reset (remove) the key that belongs to an account
```

Comment:
Thanks for adding all these comments! I don't think this one is completely accurate, since we are replacing the existing api key with a new one. 
```suggestion
    /// Reset the key that belongs to an account
```